apps/tx-builder/src/components/forms/fields/Field.tsx

```diff
@@ -0,0 +1,103 @@
+import { ReactElement } from 'react';
+import { Control, Controller } from 'react-hook-form';
+import { SelectItem } from '@gnosis.pm/safe-react-components/dist/inputs/Select';
+
+import {
+  ADDRESS_FIELD_TYPE,
+  BOOLEAN_FIELD_TYPE,
+  CONTRACT_METHOD_FIELD_TYPE,
+  HEX_ENCODED_DATA_FIELD_TYPE,
+} from './fields';
+import AddressContractField from './AddressContractField';
+import SelectContractField from './SelectContractField';
+import TextareaContractField from './TextareaContractField';
+import TextContractField from './TextContractField';
+import validateField from '../validations/validateField';
+
+function Field({ fieldType, control, name, showField = true, shouldUnregister = true, options, ...props }: FieldProps) {
+  // Component based on field type
+  const Component = CUSTOM_SOLIDITY_COMPONENTS[fieldType] || TextContractField;
+
+  return (
+    <>
+      {showField && (
+        // see https://react-hook-form.com/advanced-usage#ControlledmixedwithUncontrolledComponents
+        <Controller
+          name={name}
+          control={control}
+          defaultValue={CUSTOM_DEFAULT_VALUES[fieldType] || ''}
+          shouldUnregister={shouldUnregister}
+          rules={{
+            required: {
+              value: true,
```

Comment:
Maybe it's best to take this from a prop so that it isn't always required? You can default it to true when you destructure it from the props though.

Reply:
Good Catch! All fields are required now, but as you mention it's best to take this from a prop, because in the future we could need it:
```
          rules={{
            required: {
              value: required,
              message: 'Required',
            },
            valida
```